### PR TITLE
MAINT/STY: fft: remove E501 (line length) lint ignore, `noqa` clean-up

### PR DESCRIPTION
**Pull request by lucascolley**
#### Reference issue
Towards gh-19479.

#### What does this implement/fix?
- Follow-up to gh-19489 for `fft`.
- Clean up a `noqa` for unused import by introducing an `__all__`.

#### Additional information
I can add in a git blame ignore if wanted once this is ready.

cc @mdhaber @j-bowhay. Feel free to wait for gh-19497 if you don't want to test locally.

### CODE REVIEW

File: scipy/fft/_backend.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def register_backend(backend):
     array([1.+0.j])
     >>> set_global_backend("scipy")  # Restore global backend to default
 
-    """
+    """  # noqa: E501
```
Comment (lucascolley) [inline scipy/fft/_backend.py:128]: any sensible change for line 124?

File: scipy/fft/_pocketfft/helper.py [hunk 2]
```diff
@@ -6,7 +6,9 @@
 
 import numpy as np
 # good_size is exposed (and used) from this import
-from .pypocketfft import good_size  # noqa: F401
+from .pypocketfft import good_size
+
+__all__ = ['good_size', 'set_workers', 'get_workers']
```
Comment (lucascolley) [inline scipy/fft/_pocketfft/helper.py:11]: Not 100% but I think this is fine?

File: scipy/fft/_backend.py (hunk 1 above)
Comment (j-bowhay) [reply scipy/fft/_backend.py:128]: Would putting the comment on its own line and split across lines if necessary?

**Comment by tylerjereddy** [comment]
@steppi have you seen the unrelated `test_intermediate_overlow` with NumPy `main`/ Python `3.12` in `special` before?

The refguide-check failure is also unrelated I think, we may need to pin the NumPy version to `1.23.x` and/or the Python version back to `3.9` (maybe trying the NumPy pin first would make sense, if the Python version matters that a bit weirder perhaps..).

**Comment by ev-br** [comment]
> refguide-check failure is also unrelated

These need some `# may vary` comments it seems.

**Review by j-bowhay — APPROVED** [review]
LMTM subject to blame ignore

**Pull request merged**